lib/proxy/log_buffer/buffer.rb

```diff
@@ -0,0 +1,83 @@
+require 'proxy/log_buffer/ring_buffer'
+
+module Proxy::LogBuffer
+
+  LogRecord = Struct.new(:serial, :timestamp, :level, :message)
```

Comment:
Why do we need a serial? Why not use a string/constant for the level of the message, instead of a cryptic integer. Not gonna insist on message classes, but having two structs probably means that you should consider that.